aries_cloudagent/core/conductor.py

```diff
@@ -108,6 +106,52 @@ async def setup(self):
         """Initialize the global request context."""
 
         context = await self.context_builder.build_context()
+        ext_plugin = context.settings.get("external_plugins")
+        ext_plugin_config = context.settings.get("plugin_config")
+        in_transport_config_exception = False
+        if not context.settings.get("transport.inbound_configs"):
```

Comment:
This does rather defeat the purpose of supporting external queue plugins. Maybe we can add a way for the plugin to indicate that it handles inbound or outbound transport messages once it's loaded? The plugin registry object could look for a method in the plugin metadata, for example.

Reply:
> How about within the plugin code we add `transport.inbound_plugin_handle` and `transport.outbound_plugin_handler` [bool] to `profile.settings`. Then modify the validation in conductor as:
> 
> ```
> if not context.settings.get("transport.inbound_configs") and not context.settings.get("transport.inbound_plugin_handle"):
>         raise ArgsParseError(
>             "No --inbound-transport/-it and external transport config specified"
>         )
> if not context.settings.get("transport.outbound_configs") and not context.settings.get("transport.outbound_plugin_handle"):
>         raise ArgsParseError(
>             "No --outbound-transport/-ot and external transport config specified"
>         )
> ```

I am leaving `-it` and `-ot` validation unchanged, with the plugin it will be something like `-it redis_queue.inbound redis 0 -oq redis_queue.outbound redis 0` or `-it kafka_queue.inbound kafka 0 -oq kafka_queue.outbound kafka 0`